### ISSUE TITLE
Issue with local Validator node bring up

### ISSUE THREAD

**Issue by Keerthana-Bidarakoppa** [tablelandnetwork/local-tableland#83]
**Describe the issue**

Trying to bring up local-instance of Validator node results in error -

**To Reproduce**

From the docker directory run:
`make local-up`

**Error**
`Attaching to docker_api_1
api_1  | API server listening at: [::]:40000
api_1  | 2022-11-01T05:00:45Z warning layer=rpc Listening for remote connections (connections are not authenticated nor encrypted)
api_1  | file:///root/.tableland/database.db?_busy_timeout=5000&_foreign_keys=on&_journal_mode=WAL5:00AM INF database migration executed chain_id=31337 component=systemstore dbVersion=3 dirty=false goversion=go1.19 severity=Info version=74c5a4e
api_1  | 5:00AM FTL spinning up chain stack error="failed to connect to ethereum endpoint: dial tcp 127.0.0.1:8545: connect: connection refused" chain_id=31337 goversion=go1.19 severity=Alert version=74c5a4e
api_1  | 2022-11-01T05:00:48Z error layer=rpc writing response:write tcp 127.0.0.1:40000->127.0.0.1:47438: use of closed network connection
`

**Comment by Keerthana-Bidarakoppa** [comment]
@joewagner Thanks for the prompt response.

My system is sitting behind a firewall. Both port 8080 and 8545 are open and in listening state when I launch the service. 
I am not making any modifications in the docker file and launching the service as given in the instructions.

- Operating System - Ubuntu 20.04
- npm version - 8.11.0
- node.js version - v16.16.0
- local-tableland version - 0.0.0-pre12

**Comment by Keerthana-Bidarakoppa** [comment]
@joewagner Tried with local-tableland v0.0.2 and v0.0.0-pre13.9

Looks like the issue is not yet resolved

[Validator] Attaching to docker_api_1
[Validator] api_1  | 2022-11-14T13:44:55Z warning layer=rpc Listening for remote connections (connections are not authenticated nor encrypted)
[Validator] api_1  | API server listening at: [::]:40000
[Validator] api_1  | 1:44PM INF database migration executed error="no migration" chain_id=31337 component=systemstore dbVersion=0 dirty=false goversion=go1.19 severity=Info version=1bdbdb1
[Validator] api_1  | 1:44PM FTL creating chains stack error="creating chain_id=31337 stack: failed to connect to ethereum endpoint: dial tcp 127.0.0.1:8545: connect: connection refused" goversion=go1.19 severity=Alert version=1bdbdb1
[Validator] api_1  | 2022-11-14T13:44:56Z error layer=rpc writing response:write tcp 127.0.0.1:40000->127.0.0.1:58204: use of closed network connection
[Validator]

**Comment by joewagner** [comment]
@Keerthana-Bidarakoppa Looks like you have things configured to use Docker, which is no longer necessary and only useful if you are trying to contribute core.
Try doing the following:
 1.  create a new empty directory and cd into it, example: `mkdir test-local && cd test-local`
 2. run `npm install @tableland/local`
 3. run `npx local-tableland`

See if that gets you up and running.  If it doesn't paste the entire output of the console logs either here or in a github gist.

**Comment by joewagner** [comment]
@Keerthana-Bidarakoppa If you are in the Tableland Discord, feel free to open a ticket there too.

**Comment by joewagner** [comment]
I had another thought, in case you had previously installed globally do: `npm uninstall -g @tableland/local` before running the above commands.

**Comment by dtbuchholz** [comment]
@joewagner i was running into some issues and figured i'd add some comments over here.

so, i did the following to set up an empty project with `local-tableland`:
> 
> 1. create a new empty directory and cd into it, example: `mkdir test-local && cd test-local`
> 2. run `npm install @tableland/local`
> 3. run `npx local-tableland`
> 

upon running step (3), the following gets thrown (for both v0.0.2 and v0.0.0-pre13.9):
```
[Registry] Error HH14: Your Hardhat project uses typescript, but it's not installed.
[Registry] Please run: npm install --save-dev typescript
[Registry] For more info go to https://hardhat.org/HH14 or run Hardhat with --show-stack-traces
```
it initiates a cascade of missing dev dependency errors for: `typescript`, `ts-node`, `@nomicfoundation/hardhat-chai-matchers`, `@nomiclabs/hardhat-ethers`, `hardhat`, `chai`, `@nomiclabs/hardhat-etherscan`, `@typechain/hardhat`, `hardhat-gas-reporter`, `solidity-coverage`, `typechain`, and `@typechain/ethers-v5` -- thus, i installed these:
```
npm i -D typescript ts-node @nomicfoundation/hardhat-chai-matchers @nomiclabs/hardhat-ethers hardhat chai @nomiclabs/hardhat-etherscan @typechain/hardhat hardhat-gas-reporter solidity-coverage typechain  @typechain/ethers-v5
```

and...after installing, `local-tableland` works smoothly! very nice. but, any idea why these are all being required?

for reference:
- OS: Mac M1
- npm: 8.15.0
- node: v18.7.0
- local-tableland: 0.0.2

**Comment by dtbuchholz** [comment]
one random thing to note, in case others run into the issue. if you're using the SDK with `local-tableland` and don't set `rpcRelay` option to `false`, then you'll likely run into a nonce issue. since i believe relaying is being removed as part of a larger refactor, this won't be an issue in the future but just something to note in the interim:
```
Error: calling RelayWriteQuery: sending tx: retryable RunSQL call: contract call: Nonce too low. Expected nonce to be 15 but got 4.
```

**Comment by joewagner** [comment]
@dtbuchholz Thanks for the report!
I'm not sure why those devDeps are being required, but I'll look into it today.
For the nonce issue, you can also reset the nonce in your wallet.  In metamask you would do Settings -> Advanced -> Reset Account, for the hardhat account.  We will stop relaying soon, but even so we should probably mention the nonce needing to be reset for the hardhat accounts in the docs somewhere.

**Comment by dtbuchholz** [comment]
Forsure, thanks for the link. Should have googled the issue more lol

**Comment by dtbuchholz** [comment]
@joewagner hmm, still not working for me, unfortunately. i'm setting up a new mac later today so can test that out there, too.

btw, is there a local gateway set up in this process? i see the rpc requests go to `http://localhost:8080/rpc` but curious if i can access things over REST.

**Comment by joewagner** [comment]
@dtbuchholz yep the full gateway api is available.  As an example if you do `curl http://localhost:8080/chain/31337/tables/1` it will show you the healthbot table.

**Comment by Keerthana-Bidarakoppa** [comment]
@joewagner @dtbuchholz 

I was able to bring the local-tableland network up and running and was successful in dealing with Tables.
I had to clean up the previous installations of tableland and run the right node version. 
Thank you for helping me resolve the issue.

**Comment by joewagner** [comment]
@Keerthana-Bidarakoppa That's great to hear!  Let's leave this open a little longer while we try to investigate the dependencies that were not being installed for @dtbuchholz

**Comment by dtbuchholz** [comment]
@joewagner Confirmed this isn't an issue on a different Mac. Not sure what was going on with my other one, so I think it's safe to close this out, unless you think otherwise.
